plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveWriterFactory.java

```diff
@@ -570,24 +574,22 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
         };
 
         if (!sortedBy.isEmpty()) {
-            FileSystem fileSystem;
+            TrinoFileSystem fileSystem;
             Path tempFilePath;
             if (sortedWritingTempStagingPathEnabled) {
                 String tempPrefix = sortedWritingTempStagingPath.replace(
                         "${USER}",
                         new HdfsContext(session).getIdentity().getUser());
+                tempPrefix = setSchemeToFileIfOneDoesNotExist(tempPrefix);
                 tempFilePath = new Path(tempPrefix, ".tmp-sort." + path.getParent().getName() + "." + path.getName());
             }
             else {
                 tempFilePath = new Path(path.getParent(), ".tmp-sort." + path.getName());
             }
             try {
-                Configuration configuration = new Configuration(outputConf);
-                // Explicitly set the default FS to local file system to avoid getting HDFS when sortedWritingTempStagingPath specifies no scheme
-                configuration.set(FS_DEFAULT_NAME_KEY, "file:///");
-                fileSystem = hdfsEnvironment.getFileSystem(session.getIdentity(), tempFilePath, configuration);
+                fileSystem = fileSystemFactory.create(session);
             }
-            catch (IOException e) {
+            catch (RuntimeException e) {
```

Comment:
If it doesn't throw anything just remove the try/catch

Reply:
I don't want to change when specific exception is thrown, if you check deeper then you will see that I cach `IOException` there and throw `UncheckedIO`. I will change this to catch UncheckedIO only. Does it make sense ?